.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+name: CI
+on: 
+  push:
+    branches:
+      - '*'
+
+jobs:
+  test:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - name: Yarn install
+        uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: install
+      - name: Yarn Test
+        uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: test
+
+  publish-npm:
+    needs: test
+    runs-on: ubuntu-18.04
+    if: startsWith(github.ref, 'refs/tags/') 
```

Comment:
can this be moved to just the `Yarn publish` step? https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsif That way `yarn build` (including TypeScript compilation?) is always executed